doc/source/train/user-guides/data-loading-preprocessing.rst

```diff
@@ -177,25 +185,123 @@ Using Ray Data and Ray Train for distributed training on large datasets involves
+
+This behavior can be overwritten by passing in the ``dataset_config`` argument. For more information on configuring splitting logic, see :ref:`Splitting datasets <train-datasets-split>`.
+
+.. _train-datasets-consume:
 
-Some deep learning frameworks provide their own dataloading utilities. For example:
+Consuming data
+~~~~~~~~~~~~~~
 
-- PyTorch: `PyTorch Dataset & DataLoader <https://pytorch.org/tutorials/beginner/basics/data_tutorial.html>`
-- HuggingFace: `HuggingFace Dataset <https://huggingface.co/docs/datasets/index>`
-- PyTorch Lightning: `LightningDataModule <https://lightning.ai/docs/pytorch/stable/data/datamodule.html>`
+Inside the ``train_loop_per_worker``, each worker can access its shard of the dataset via :meth:`ray.train.get_dataset_shard`.
 
-You can still use the aforementioned utilities in Ray Train. However, for more performant large-scale data ingestion, you should consider migrating to Ray Data.
+This data can be consumed in a variety of ways:
 
-PyTorch Datasets are replaced by the :class:`Dataset <ray.data.Dataset>` abtraction, and the PyTorch DataLoader is replaced by :meth:`Dataset.iter_torch_batches() <ray.data.Dataset.iter_torch_batches>`.
+- To create a generic Iterable of batches, you can call :meth:`~ray.data.DataIterator.iter_batches`.
+- To create a replacement for a PyTorch DataLoader, you can call :meth:`~ray.data.DataIterator.iter_torch_batches`.
+
+For more details on how to iterate over your data, see :ref:`Iterating over data <iterating-over-data>`.
+
+.. _train-datasets-pytorch:
+
+Starting with PyTorch data
+--------------------------
+
+Some frameworks provide their own dataset and data loading utilities. For example:
+
+- **PyTorch:** `Dataset & DataLoader <https://pytorch.org/tutorials/beginner/basics/data_tutorial.html>`_
+- **Hugging Face:** `Dataset <https://huggingface.co/docs/datasets/index>`_
+- **PyTorch Lightning:** `LightningDataModule <https://lightning.ai/docs/pytorch/stable/data/datamodule.html>`_
+
+These utilities can still be used directly with Ray Train. In particular, you may want to do this if you already have your data ingestion pipeline set up.
+However, for more performant large-scale data ingestion we do recommend migrating to Ray Data.
+
+At a high level, you can compare these concepts as follows:
+
+.. list-table::
+   :header-rows: 1
+
+   * - PyTorch API
+     - HuggingFace API
+     - Ray Data API
+   * - `torch.utils.data.Dataset <https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset>`_
+     - `datasets.Dataset <https://huggingface.co/docs/datasets/main/en/package_reference/main_classes#datasets.Dataset>`_
+     - :class:`ray.data.Dataset`
+   * - `torch.utils.data.DataLoader <https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader>`_
+     - n/a
+     - :meth:`ray.data.Dataset.iter_torch_batches`
+
+
+For more details, see the following sections for each framework.
+
+.. tabs::
+
+    .. tab:: PyTorch Dataset and DataLoader
```

Comment:
Should we also mention `ray.data.from_torch` here? If they want to quickly convert their small torch datasets.

Reply:
I didn't put it there because it's described in the `Migrating from PyTorch Datasets and DataLoaders` page that's linked

Reply:
Got it. Make sense:)